diff --git a/src/nodes/btree.rs b/src/nodes/btree.rs
index 21f2f1e..d501f14 100644
--- a/src/nodes/btree.rs
+++ b/src/nodes/btree.rs
@@ -446,7 +446,7 @@ impl<A: BTreeValue> Node<A> {
                     path
                 }
                 None => {
-                    path.push((self, index - 1));
+                    path.push((self, index));
                     path
                 }
                 Some(ref node) => {
